frame/staking/src/tests.rs

```diff
@@ -1863,39 +1864,39 @@ fn bond_with_duplicate_vote_should_be_ignored_by_npos_election_elected() {
 		.execute_with(|| {
 			// disable the nominator
 			assert_ok!(Staking::chill(Origin::signed(100)));
-			// make stakes equal.
+			// 31/30 will have less stake
 			assert_ok!(Staking::bond_extra(Origin::signed(31), 99));
-
+			// ensure all have equal stake.
 			assert_eq!(
 				<Validators<Test>>::iter()
 					.map(|(v, _)| (v, Staking::ledger(v - 1).unwrap().total))
 					.collect::<Vec<_>>(),
 				vec![(31, 100), (21, 1000), (11, 1000)],
 			);
+			// no nominators shall exist.
 			assert!(<Nominators<Test>>::iter().map(|(n, _)| n).collect::<Vec<_>>().is_empty());
 
-			// give the man some money
+			// give the man some money.
 			let initial_balance = 1000;
 			for i in [1, 2, 3, 4,].iter() {
 				let _ = Balances::make_free_balance_be(i, initial_balance);
 			}
 
 			assert_ok!(Staking::bond(Origin::signed(1), 2, 1000, RewardDestination::Controller));
-			assert_ok!(Staking::nominate(Origin::signed(2), vec![11, 11, 11, 21, 31,]));
+			assert_ok!(Staking::nominate(Origin::signed(2), vec![11, 11, 11, 21, 31]));
 
 			assert_ok!(Staking::bond(Origin::signed(3), 4, 1000, RewardDestination::Controller));
 			assert_ok!(Staking::nominate(Origin::signed(4), vec![21, 31]));
 
-			// winners should be 21 and 31. Otherwise this election is taking duplicates into account.
-			let sp_npos_elections::ElectionResult {
-				winners,
-				assignments,
-			} = Staking::do_phragmen::<Perbill>(0).unwrap();
-
-			let winners = sp_npos_elections::to_without_backing(winners);
-			assert_eq!(winners, vec![21, 11]);
-			// only distribution to 21 and 31.
-			assert_eq!(assignments.iter().find(|a| a.who == 1).unwrap().distribution.len(), 2);
+			// winners should be 21 and 11.
```

Comment:
Why did the winner change in this test?

Reply:
it actually hasn't. Previously it was 11, 21, and it still is. Only the doc was wrong.

Reply:
Comparing to the previous test, this time 31 has less stake, so 11 who has dupe votes is in the winner set. But we see that the `Support.voters` is coming back sane and the dupes have been stripped away.